### PR TITLE
Use email module to work and decode email in OOP and generic way.

### LINKED ISSUE #22: Fix decoding problems once and for all
If email payload is base64 encoded or if email is encoded using different encoding than UTF-8 or ISO-8859-1 method get_email_body return's not correctly decoded body. Solution is instead use email module, which can parse/decode whole email and gives ability to access properties and methods OOP way.

Fix was inspired by https://www.thepythoncode.com/article/reading-emails-in-python .

### PR DESCRIPTION
**Pull request by navrkald**
Fixes #22.

### CODE REVIEW

**Comment by lasselindqvistsolibri** [comment]
Thanks! How did you test this?

File: src/ImapLibrary2/__init__.py [hunk 1]
```diff
@@ -132,11 +132,18 @@ def get_email_body(self, email_index):
...
+            msg = message_from_bytes(encodedMessage)
+            if not msg.is_multipart():
+                body = msg.get_payload(decode=True).decode()
+            else:
+                # decode the email subject
+                subject, encoding = decode_header(msg["Subject"])[0]
+                if isinstance(subject, bytes):
+                    # if it's a bytes, decode to str
+                    subject = subject.decode(encoding)
+                raise Exception("get_email_body called on multipart email '%s'. Pls first use method walk_multipart_email." % (subject))
```
Comment (lasselindqvistsolibri) [inline src/ImapLibrary2/__init__.py]: Could replace "Pls" with "Please".

Comment (navrkald) [reply src/ImapLibrary2/__init__.py]: Thanks fixed.

**Comment by navrkald** [comment]
> Thanks! How did you test this?

I have written simple code to loop over all emails in my mailbox and manually checked encoding. Something like this:

```python
username = "<my email address>"
access_token = "<my OAuth2 access token>"

imapLibObj = ImapLibrary2()

imapLibObj.open_mailbox_oauth (host="outlook.office365.com", debug_level=2, user=username, access_token=authToken)
imapLibObj._get_all_emails()
for num in imapLibObj._mails:
    try:
        body = imapLibObj.get_email_body(num)
        print(body)
    except Exception as err:
        print(err)

imapLibObj.close()
imapLibObj.logout()
```

**Comment by lasselindqvist** [comment]
Thanks! I will try to find time to make a release out of this during the weekend.

**Comment by navrkald** [comment]
Hello @lasselindqvist , thanks for merging PR. Is there please any chance, new version will be released soon?

**Comment by lasselindqvist** [comment]
> Hello @lasselindqvist , thanks for merging PR. Is there please any chance, new version will be released soon?

Hi. I released this in 0.4.2 now. Thanks for the fix.

**Comment by navrkald** [comment]
Np and thanks for releasing!

**Pull request merged**